app/views/snacks/show.html.erb

```diff
@@ -0,0 +1,8 @@
+<h1><%= @snack.name %></h1>
+<p>Price: $<%= @snack.price %></p><hr>
+<h3>Locations</h3>
+  <ul>
+    <% @snack.machines.each do |machine| %>
+      <li><%= machine.location %> (<%= machine.snacks.count %> kinds of snacks, average price of $<%= machine.avg_snack_price %>)</li>
```

Comment:
`machine.snacks.count` --- I would create a instance method for Machine that counted all of the snacks. This is logic that we want to take out of the view.